### PR TITLE
Feature/amltk converter

### PR DESCRIPTION
**Pull request by sarah-segel**
## Converters
- Add support for AMLTK.

Open points:

- For Optuna, I partly get 2 entries with the same config values (and same seed) - not sure what is the reason for this? With the current converter implementation, only one of those 2 entries is considered in DeepCave in this case.
- I am not sure what is the best way to extract budget, as there is not fidelity/budget column. Currently the converter is taking it from the "name" column, so this would only work for smac. Another option would be to ask the user to add it in the history.csv themselves during the AMLTK run (I assume via `trial.summary["budget"] = trial.fidelities["budget"] `) and read it from there?

### CODE REVIEW

**Comment by eddiebergman** [comment]
1. No idea why Optuna is doing that to be honest. The `name` field is meant to uniquely define a single configuration but perhaps there was some information I did not capture in there...
2. There is a "budget" column used, but **only** in the case of multi-fidelity (which is rather undocmented/tested). My recommendation is that if there is no budget column, then to use some constant column.

File: deepcave/runs/converters/amltk.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                        lower=lower,
+                        upper=upper,
+                        optimize="upper" if maximize else "lower",
+                    )
+                )
+
+        obj_list.append(Objective("Time"))
+
+        # Let's create a new run object
+        run = AMLTKRun(name=path.stem, configspace=configspace, objectives=obj_list, meta=None)
```
Comment (eddiebergman) [inline deepcave/runs/converters/amltk.py:134]: Technically anything under `summary:` could be considered user provided `meta=`

File: docs/converters.rst [hunk 2]
```diff
@@ -12,13 +12,53 @@ Currently, DeepCAVE supports the following converters:
...
+
+To be able to load your AMLTK run into DeepCave, there are a few points to
+consider when running AMLTK:
+
+1. Save the ``trial_history`` as a CSV file:
+
+   .. code-block:: python
+
+      history_df = trial_history.df()
+      bucket.store({"history.csv": history_df})
```
Comment (eddiebergman) [inline docs/converters.rst:34]: Same here with `parquet`.

File: deepcave/runs/converters/amltk.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        Returns
+        -------
+        str
+            The hash of the run.
+        """
+        if self.path is None:
+            return ""
+
+        # Use hash of history.csv as id
+        return file_to_hash(self.path / "history.csv")
```
[the remark this answers is not in the document]
Comment (sarah-segel) [reply deepcave/runs/converters/amltk.py]: Good point, replaced csv with parquet everywhere.

File: deepcave/runs/converters/amltk.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+        with open(path / "configspace.json", "rb") as f:
+            json_string = pickle.load(f)
+        configspace = cs_json.read(json_string)
+
+        history = pd.read_csv(path / "history.csv")
+
+        history["budget"] = history["name"].apply(
+            lambda x: float(value) if (value := extract_value(x, "budget")) is not None else None
+        )
+        history["instance"] = history["name"].apply(lambda x: extract_value(x, "instance"))
```
[the remark this answers is not in the document]
Comment (sarah-segel) [reply deepcave/runs/converters/amltk.py]: Good to know, I removed that - was only using it to check whether there are multiple instances anyway (which cannot be handled in DeepCave at the moment).

File: deepcave/runs/converters/amltk.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+        # Extract the objectives from the dataframe
+        obj_list = list()
+        for metric_string in history.columns:
+            if metric_string.startswith("metric:"):
+                match = re.match(
+                    r"metric:(\w+) \[(\d+\.\d+), (\d+\.\d+)\] \((\w+)\)", metric_string
```
[the remark this answers is not in the document]
Comment (sarah-segel) [reply deepcave/runs/converters/amltk.py]: True. Edited the regex such that now it should work for floats/ints and +-np.inf.

File: deepcave/runs/converters/amltk.py (hunk 1 above)
Comment (sarah-segel) [reply deepcave/runs/converters/amltk.py:134]: This is meta information per trial though, right? I did not add it as we currently do not make use of per-trial meta information (only information on the overall run, such as SMAC version, facade etc).

**Pull request merged**